src/test/pmemset_sds/pmemset_sds.c

```diff
@@ -0,0 +1,456 @@
+	ret = pmemset_config_set_acceptable_states(cfg, acceptable_states);
+	UT_PMEMSET_EXPECT_RETURN(ret, 0);
+
+	ret = pmemset_new(&set, cfg);
+	UT_PMEMSET_EXPECT_RETURN(ret, 0);
+
+	sds.refcount = 0;
+	extras.sds = &sds;
+	extras.bb = NULL;
+	extras.state = &state;
+	pmemset_source_set_extras(src, &extras);
+
+	/* get extras with duplicated SDS (internal function) */
+	extras = pmemset_source_get_extras(src);
+	/* spoil device id */
+	extras.sds->id[0] += 1;
+
+	char old_device_id[PMEMSET_SDS_DEVICE_ID_LEN];
+	strncpy(old_device_id, extras.sds->id, PMEMSET_SDS_DEVICE_ID_LEN);
+
+	ut_create_map_config(&map_cfg, set, 0, 0);
+
+	/* new SDS unsafe shutdown count doesn't match the old one */
+	ret = pmemset_map(src, map_cfg, NULL);
+	if (ret == PMEMSET_E_SDS_ENOSUPP)
+		goto err_cleanup;
+	UT_PMEMSET_EXPECT_RETURN(ret, 0);
+	UT_ASSERTeq(state, PMEMSET_PART_STATE_OK);
+	/* if pool wasn't in use then device id should be reinitialized */
+	UT_ASSERT(strncmp(extras.sds->id, old_device_id,
+			PMEMSET_SDS_DEVICE_ID_LEN) != 0);
+
+err_cleanup:
+	ret = pmemset_delete(&set);
+	UT_PMEMSET_EXPECT_RETURN(ret, 0);
+	ret = pmemset_config_delete(&cfg);
+	UT_PMEMSET_EXPECT_RETURN(ret, 0);
+	ret = pmemset_map_config_delete(&map_cfg);
+	UT_PMEMSET_EXPECT_RETURN(ret, 0);
+	ret = pmemset_source_delete(&src);
+	UT_PMEMSET_EXPECT_RETURN(ret, 0);
+
+	return 1;
+}
+
+/*
+ * test_sds_pool_multiple_mappings - create new sds and map three parts one by
+ *                                   one
+ */
+static int
+test_sds_pool_multiple_mappings(const struct test_case *tc, int argc,
+		char *argv[])
+{
+	if (argc < 1)
+		UT_FATAL("usage: test_sds_pool_normal_mappings " \
+				"<path>");
+
+	const char *file = argv[0];
+	struct pmemset *set;
+	struct pmemset_config *cfg;
+	struct pmemset_map_config *map_cfg;
+	struct pmemset_extras extras;
+	enum pmemset_part_state state;
+	struct pmemset_sds sds;
+	struct pmemset_source *src;
+
+	int ret = pmemset_source_from_file(&src, file);
+	UT_PMEMSET_EXPECT_RETURN(ret, 0);
+	UT_ASSERTne(src, NULL);
+
+	ut_create_set_config(&cfg);
+
+	uint64_t acceptable_states = (PMEMSET_PART_STATE_OK |
+			PMEMSET_PART_STATE_OK_BUT_INTERRUPTED);
+	ret = pmemset_config_set_acceptable_states(cfg, acceptable_states);
+	UT_PMEMSET_EXPECT_RETURN(ret, 0);
+
+	ret = pmemset_new(&set, cfg);
+	UT_PMEMSET_EXPECT_RETURN(ret, 0);
+
+	sds.refcount = 0;
+	extras.sds = &sds;
+	extras.bb = NULL;
+	extras.state = &state;
+	pmemset_source_set_extras(src, &extras);
+
+	ut_create_map_config(&map_cfg, set, 0, 0);
+
+	ret = pmemset_map(src, map_cfg, NULL);
+	if (ret == PMEMSET_E_SDS_ENOSUPP)
+		goto err_cleanup;
+	UT_PMEMSET_EXPECT_RETURN(ret, 0);
+	UT_ASSERTeq(state, PMEMSET_PART_STATE_OK);
+
+	ret = pmemset_map(src, map_cfg, NULL);
+	UT_PMEMSET_EXPECT_RETURN(ret, 0);
+	/*
+	 * sds state changes after first mapping, (it's in use), therefore every
+	 * next mapping from this SDS should return an interrupted state
+	 */
+	UT_ASSERTeq(state, PMEMSET_PART_STATE_OK_BUT_INTERRUPTED);
```

Comment:
I'm okay with this behavior, but this should probably be either its own state (OK_BUT_ALREADY_OPEN or something) or explained in the man page.